include/swift/Basic/ArrayRefView.h

```diff
@@ -25,15 +25,17 @@ namespace swift {
 /// An adapter for iterating over a range of values as a range of
 /// values of a different type.
 template <class Orig, class Projected, Projected (&Project)(const Orig &),
-          bool AllowOrigAccess = false>
+          bool AllowOrigAccess = false, class ArrayRefTy = llvm::ArrayRef<Orig>>
```

Comment:
I'm not happy with this change; one feature of ArrayRefView is that copying it has to be just like copying ArrayRef. Can you use TransformRange instead?

Reply:
I am fine with that.